### DISCUSSION TITLE
Direction for Elementary OS

### DISCUSSION THREAD

**Discussion by rwoolf** [elementary/triage#174] [Ideas]
I have some thoughts for the direction of Elementary OS that I would like to share.  I have been using Elementary OS as my daily driver work OS since Luna.  I am a developer and since 2013 I have done all of my development work in Elementary OS.  So what I say comes with some experience.  I use Virtual Machines to do all of my development work in.  This allows me to have many Virtual Machines with different environments or versions of our applications that have been customized for our customers.  I can easily take snapshots before making major changes and if things go awry it is just a few seconds to be back to a good state of things. Or I can make a branch as I experiment with ideas. There are a lot of advantages for developing in a virtual machine.  So while my host machine is Windows, provided by my employer, all of my development virtual machines have been Elementary OS, which I adopted those many years ago because of its clean UI. I have promoted the OS to others over all of these years. But I have always had to tweak it a bit to fit my needs.  

The reality is that the majority of the world primarily uses two OS's on a desktop or laptop, Windows and Mac, with Chrome being a distant third.  Elementary OS promotes and is designed to be a simple OS for the average person and promotes the default simple apps of browser, media, email, etc.  But the majority of those level of users are going to stick with the OS that came on the laptop they bought. That level of users are not adventurous enough take on a "Linux" themselves. It is higher level users that will, and this is where Elementary OS comes in deficient.  I propose that gearing Elementary OS to developers will get a far broader adoption, and the spill over to other users will happen as they promote it to their family and friends.

The following are my thoughts on the topic.
- The single click for files and folders was problematic.  I see that with 7 this has been fully resolved.  Thanks.  Anyone coming from another OS will have the muscle memory and being able to transition back and forth is of great benefit.  I used the Tweaks app to modify this for prior versions.  I simply bring this up because being able to transition between OS's is important for anyone to adopt Elementary OS.  They need to feel at home and not have to learn a new way of doing standard things.
- Another thing I tweak is to add the minimize button on.  I understand trying to promote a philosophy, but like the double click of the prior point, anyone coming from another OS (and likely moving back and forth) would like the window control icons to be the same, and even in the same place as their other primary OS.  This is really such an insignificant thing, allowing someone the choice of how they want their buttons.  It does not invalidate Elementary OS to allow this.  It should not require someone to install the tweaks app to make this happen.  For every person I have helped set up on Elementary OS, they have not liked not having the minimize button as that is what their muscle memory expects when working with windows.  I then show them tweaks and they always choose to setup the icons like the OS they are coming from.  I don't think Elementary OS should hinder its own adoption by such an insignificant thing as allowing a user to adopt their choice of window controls.  Allowing choice will only help adoption.
- Broaden back up the way to install apps.  I understand wanting to sandbox apps through Flatpak, but this is still too limiting and restrictive.  You should easily allow people to access the older deb packed apps in the App Center that we last had in Elementary OS 5.  While those who know how can still install deb apps via the command line on 7, having a nice clean and simple way to install a broader range of apps from the App Center is preferred.  It is okay to promote Flatpak over the other installs, and curated vs. non.  And even warn of potential instability when they are installing deb packages from the App Center, but keep it as simple as possible for people to find the things they need to do their work.
- In similar fashion to the above, I have a few apps that are windows only apps that our organization has standardized on for development, and I have installed them in Elementary OS via Wine.  In Elementary OS 5 there was a wonderful curated app named Wineglass that made using Wine on Elementary OS a breeze and very simple.  But the Flatpak packaging requirements squashed that valuable app.  But it or another similar app should be easily available to help people run those critical windows apps they may need.
- Provide a way in the app center to have a one click install of Docker and tools for working with Docker, and other increasingly important tools for the developer community.  Make it easy to be able to do their development work.
- Make Elementary OS a first class citizen when running in a Virtual Machine.  Promote and help people understand how they can use a virtual machine to use Elementary OS side by side with their host OS of choice.  Help developers see how beneficial it is to develop in virtual machines as well as the benefits of using Elementary OS as their development OS in these virtual machines. Maybe provide an OVF image of a pre-installed Elementary OS that on first boot lets them configure and setup user, etc.  This would be very quick and easy for someone to get up and going in a VM. In this scenario, developers don't have to give up their primary OS (often dictated by their employers), they simply enhance their development experience and environment by being able to have as may Elementary OS VMs as they want to help them do their work faster and easier.

I have a lot more I can offer in regards to my thoughts.  But the short of it is this.  If you geared Elementary OS as the OS of choice for developers through its clean and easy to use UI, but putting the power of what they need to do just a few clicks away, and making it easy to use this in a VM, it could see broad adoption by the development community.  Who better to have as your primary user than the people who can contribute and help fix and improve the OS.  I think many developers would appreciate a clean and elegant OS for doing their development work in.  This takes nothing away from still providing a clean  and simple OS for the average user, but some basic tweaks and direction will make it easier to adopt by either community of users.

**Comment by jeremypw** [comment]
I can see the potential of increasing the user base of providing some transitional/fallback settings to aid users migrating from other OSes or dual-working with them, especially where the settings exist but are hidden and the support cost is minimal.  The default settings should remain in line with the designer's vision though. An attempt to restore the minimize button was made a few years ago (see https://github.com/elementary/switchboard-plug-pantheon-shell/pull/145) but was rejected.  I can see that having a variable minimum width of the header if the button were optional might conflict with being able to half-tile in a few cases.   It is possible to activate a setting that minimizes the window on middle-clicking the headerbar (not currently exposed in the UI) but unless Windows has a similar option that does not help much I guess.  You can still minimize the window by clicking on the dock panel and using the short cut `<Super>H`.

I am less convinced about maintaining `.deb` support in AppCenter. The support burden is significant and having two fundamentally different types of app could confuse inexperienced users.  It could be argued that a user experienced enough to cope with two types of app would be able to install 'synaptic` or use the commandline.  Hopefully as time goes on, the number of apps not available as Flatpak will become insignificant.

The suggestion to target ElementaryOS in VMs is interesting and could attract more developers to the project.

Note that these are my views, not those of ElementaryOS.

**Reply by jhaygood86** [reply]
I agree with the .deb support -- applications installed as flatpak generally provide a much better user experience on elementary, since either they are A) elementary applications that fully comply with the elementary HIG and thus look and work great or B) applications for other DEs with their own HIG, and thus will look and work correctly in the style they are expecting. Also, things like automatic (and faster) updates, more "new" versions (don't have to wait on a new elementary OS release to get a new app version)

Of the major desktop apps out there, I can't think of any that aren't available in a flatpak already.

**Reply by rwoolf** [reply]
In elementary 5 you could go to the App Center and install the OpenConnect Client for Cisco AnyConnect SSL VPN for NetworkManager with a single click.  You will never get things like this from a flatpak.  

As I tried to emphasize, if the goal of Elementary OS is to provide an OS geared towards the user who only wants to do minimal things like surf the web, listen to music, or look at photos or video, then I think you will face a challenge as that type of user will likely just stick with the OS that comes with the laptop they buy.  If the answer is that a more technical person can just use the command line to install things, then they can do all of that with any Linux, so why would they go with something that restricts them so much "out of the box."  However, if you created an OS, geared towards the more technical developer community, but made it simple and elegant to deal with what they need to to do their job, then you have a chance of growing the base of users.  You grow the user base with those who can actually help to improve it.

I work with a lot of developers who stick to Windows because they don't want to invest the time and effort to learn Linux.  But if they could see how easy an alternative OS could be, and everything they need is a simple click away in the App Center, then this might be of interest to them.  Install Java OpenJDK with a single click in the app center. Install Mono or .Net., Docker, etc. from the app center.  Install all my dev apps with a single click.  Combine the elegance and simplicity, with being able to work in VMs and you have a very powerful offer to a large community of developers that will improve and promote the OS.  Many developers have moved to Mac because they like the clean simplicity of it.  Elementary OS should, in my opinion, become that same option for developers on both Windows and Mac, while working in VMs.

**Comment by jeremypw** [comment]
There are often less well known Flatpak apps doing the same job when a "traditional" app is missing (e.g. `grsync`) but without a rating system it is hard to know how good they are.  See https://sourceforge.net/p/grsync/feature-requests/22/

**Comment by danirabbit** [comment]
Gonna close this since there hasn’t been any movement in a while and it contains lots of separate ideas all in one post, which makes it really hard to tell if it has been resolved or not.

please feel free to reopen separate individual posts for things that haven’t been answered/resolved. Thanks!
